### DISCUSSION TITLE
Help integrating custom radio with React-Hook-Form (RHF)

### DISCUSSION THREAD

**Discussion by paambaati** [chakra-ui/chakra-ui#3724] [Q&A] [answered]
I'm trying to finally (whew!) upgrade my v0.8x Chakra-UI project, and I need to build a custom radio group component that also needs to submit values to [RHF v6](https://react-hook-form.com/get-started).

I have a CodeSandbox forked from the RHF example in the Guides section, and I'm having trouble getting a custom radio group to register with RHF.

https://codesandbox.io/s/chakra-ui-react-hook-form-custom-radio-2un9h?file=/src/HookForm.js

Has anyone had success getting these 2 woking together?

CC: @bluebill1049 for help.

**Comment by bluebill1049** [comment]
have a look at this codesandbox (it's on v7) but should be similar to v6: https://codesandbox.io/s/chakra-ui-5mp8g

**Comment by paambaati** [comment, marked as answer]
This response has a working example — https://github.com/react-hook-form/react-hook-form/discussions/4591#discussioncomment-563645
